packages/components/src/components/hds/dialog-primitive/header.ts

```diff
@@ -4,17 +4,32 @@
  */
 
 import Component from '@glimmer/component';
+import type { FlightIconSignature } from '@hashicorp/ember-flight-icons/components/flight-icon';
+
+export interface HdsDialogPrimitiveHeaderSignature {
+  Args: {
+    contextualClassPrefix?: string;
+    id?: string;
+    tagline?: string;
+    icon?: FlightIconSignature['Args']['name'] | false;
+    onDismiss?: () => void;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLDivElement;
+}
 
 const NOOP = () => {};
 
-export default class HdsDialogPrimitiveHeaderComponent extends Component {
+export default class HdsDialogPrimitiveHeaderComponent extends Component<HdsDialogPrimitiveHeaderSignature> {
   /**
    * @param onDismiss
    * @type {function}
    * @default () => {}
    */
   get onDismiss() {
```

Comment:
see for example https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/alert/index.ts#L136
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  get onDismiss(): ((event: MouseEvent, ...args: any[]) => void) | false {
```

Reply:
For this instance, I don't think I need to include false because it always returns a function. If onDismiss isn't passed, it just returns NOOP.